x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/assembler/BoxedQueryRequest.java

```diff
@@ -58,6 +58,15 @@ public BoxedQueryRequest(QueryRequest original, String timestamp, List<String> k
         timestampRange = rangeQuery(timestamp).timeZone("UTC").format("epoch_millis");
         keys = keyNames;
         RuntimeUtils.addFilter(timestampRange, searchSource);
+        // do not join on null values
+        if (keyNames.isEmpty() == false) {
+            BoolQueryBuilder nullValuesFilter = boolQuery();
+            for (int keyIndex = 0; keyIndex < keyNames.size(); keyIndex++) {
+                // add an "exists" query for each join key to filter out any non-existent values
+                nullValuesFilter.must(existsQuery(keyNames.get(keyIndex)));
```

Comment:
No need to score, just use filter. Which removes the need for wrapping it into a bool query and simply use `RuntimeUtils.addFilter` for every existsQuery.